src/components/Table/SimpleTable.js

```diff
@@ -55,12 +59,23 @@ class SimpleTable extends React.Component {
     renderDetails: null,
     columns: null,
     onSortChanged: () => null,
+    getId: undefined,
     loading: false,
   };
 
   state = {
     sortOrders: {},
-    detailsItem: null,
+    detailsItemId: null,
+  };
+
+  getId = (item) => {
```

Comment:
You can just provide this function as the `defaultProps` value for `getId` and skip the try / fallback logic.